code/synchronization/forward-trigger-service.py

```diff
@@ -127,25 +119,43 @@ async def forward_signals(serial_port: str, async_q: janus.AsyncQueue[int]) -> N
 def _trigger(sync_q: janus.SyncQueue[int]) -> None:
     """
     Trigger the signal sync queue.
-
     Put a signal into the sync queue.
-
     Parameters:
         sync_q (janus.SyncQueue[int]): The sync queue for triggering the signal.
-
     """
     sync_q.put(b"\x01")
     sync_q.join()
     logging.info("Scanner trigger received")
 
 
+def check_usb_device_connected(usb_vendor_id, usb_product_id):
+    """
+    Check if a USB device with the specified vendor and product IDs is connected.
+    Parameters:
+        usb_vendor_id (str): The USB vendor ID.
+        usb_product_id (str): The USB product ID.
+    Returns:
+        bool: True if the device is connected, False otherwise.
+    """
+    device = usb.core.find(
+        idVendor=int(usb_vendor_id, 16), idProduct=int(usb_product_id, 16)
+    )
+    if device is None:
+        raise RuntimeError(
+            f"USB device (Vendor ID: {usb_vendor_id}, Product ID: {usb_product_id}) is not connected."
+        )
+    return True
+
+
 async def main() -> None:
     """
     Main function.
-
     Run the signal server and forward the signals to the serial port.
-
     """
+    # Check if the USB device is connected
+    initial_usb_connection = check_usb_device_connected(
```

Comment:
Why keeping a variable if this is expected to crash if the device is not present?

```suggestion
    ensure_usb_device_connected(
```